### PR TITLE
Fix broken hyperlink on hover

### PR DESCRIPTION
**Pull request by michaellzc**
;) there's an extra parenthesis at the end

CleanShot 2022-02-18 at 20 40 43@2x

### CODE REVIEW

**Review by bobheadxi — APPROVED** [review]
im only a code dumb but this looks right to me

**Comment by michaellzc** [comment]
> im only a code dumb but this looks right to me

I was browsing linkerd2 codes and got annoyed by the fact there's no precise code intelligence. Clicking on the hover, but the link is broken and took me a while to spot the extra `)` at the end.

Finally, have the right link and try to run `lsif-go` to index their stuff, but it failed 🤷‍♀️

**Comment by bobheadxi** [comment]
> Finally, have the right link and try to run lsif-go to index their stuff, but it failed 🤷‍♀️

Time for another PR? 😲

**Comment by bobheadxi** [comment]
Tangent, but:

> I was browsing linkerd2 codes and got annoyed by the fact there's no precise code intelligence. ... Finally, have the right link and try to run lsif-go to index their stuff, but it failed 🤷‍♀️

I realized I've never tried an LSIF upload before! So I took the linked docs for a spin, which worked locally, but looks like you need a token with access to the repo to publish indexes (makes sense). 

So I found the auto-indexing job for `linkerd2` to try and see what's up: https://sourcegraph.com/github.com/linkerd/linkerd2/-/code-intelligence/indexes/TFNJRkluZGV4OjE1MjU4NjY=

```
jaeger/cmd/install.go:11:2: //go:build comment without // +build comment
multicluster/cmd/allow.go:8:2: //go:build comment without // +build comment
pkg/charts/charts.go:10:2: //go:build comment without // +build comment
viz/cmd/install.go:15:2: //go:build comment without // +build comment
```

A diff search for `// +build` changes reveals https://sourcegraph.com/github.com/linkerd/linkerd2/-/commit/e54061b61f7faf83cc50bd32790d07204d5ce441, indicating this removal was tied to an upgrade to go 1.17. And sure enough, `lsif-go`'s Go version is pinned to 1.16: https://github.com/sourcegraph/lsif-go/blob/master/Dockerfile#L3 , which would explain why the LSIF indexing works locally since I'm on Go 1.17

So I went into [the configuration policy](https://sourcegraph.com/github.com/linkerd/linkerd2/-/code-intelligence/index-configuration), generated a default one, and changed the Go job to use a 1.17 base image and install + run `lsif-go` on the fly:

```json
{
	"shared_steps": [],
	"index_jobs": [
		{
			"steps": [
				{
					"root": "",
					"image": "golang:1.17-buster",
					"commands": [
						"go mod download"
					]
				}
			],
			"local_steps": [],
			"root": "",
			"indexer": "golang:1.17-buster",
			"indexer_args": [
				"go install github.com/sourcegraph/lsif-go/cmd/lsif-go@latest && lsif-go --no-animation",
			],
			"outfile": ""
		},
```

And enqueued a new job: https://sourcegraph.com/github.com/linkerd/linkerd2/-/code-intelligence/indexes/TFNJRkluZGV4OjE1Mjc2NzQ=

And it worked!

image

https://sourcegraph.com/github.com/linkerd/linkerd2/-/code-intelligence/uploads/TFNJRlVwbG9hZDoxMjIzMTMx

image

tl;dr precise code intel now available for linkerd2 codes @michaellzc  😁 

@sourcegraph/code-intel this might be worth considering to make sure the autoindexing Go version matches the project's Go version, rather than relying on a publish of `sourcegraph/lsif-go`

some other thoughts:

- I must say all this was a really nice experience! though it would be nice if I could push my jobs to the front of the queue so I don't have to wait for ~30 jobs to finish
- should I be allowed to put a whole string of commands like this into `indexer_args`? I thought it might require me to pass in argument by argument, but I'm glad it let me 😛 
- manually enqueued jobs seem to get killed and replaced by jobs triggered from elsewhere

**Comment by michaellzc** [comment]
Sharing my perspective as a go newbie

I started from this page https://docs.sourcegraph.com/code_intelligence/how-to/adding_lsif_to_many_repos

It asked me to create [batch change](https://raw.githubusercontent.com/sourcegraph/snippets/main/lsif/lsif-go.campaign.yaml), okay...

Looked like I just need to pull the `sourcegraph/lsif-go` image and run `lsif-go` to index a go project, that's easy!

I cloned the linkerd2 and mount the repo into a  `sourcegraph/lsif-go` container, then just run `lsif-go`, but I kept on getting

```
root@0542754ac3c1:/app# lsif-go
✔ Resolving module name... Done
✔ Listing dependencies... Done

error: failed to list project dependencies: failed to list project packages: exit status 1
jaeger/cmd/install.go:11:2: //go:build comment without // +build comment
multicluster/cmd/allow.go:8:2: //go:build comment without // +build comment
pkg/charts/charts.go:10:2: //go:build comment without // +build comment
viz/cmd/install.go:15:2: //go:build comment without // +build comment
```

I tried looking up a few exiting issues but can't seem to find anything that's related.

**Review by Strum355 — APPROVED** [review]
😎

**Review by mrnugget — APPROVED** [review]
